app/auth/views.py

```diff
@@ -0,0 +1,198 @@
+
+@auth_blueprint.route('/login', methods=['POST'])
+def login_user():
+    """Loging in a user
+    ---
+    tags:
+     - "auth"
+    parameters:
+        - in: "body"
+          name: "body"
+          description: "Username and password"
+          required: true
+          schema:
+            type: "object"
+            required:
+            - "username"
+            - "password"
+            properties:
+                username:
+                    type: "string"
+                password:
+                    type: "string"
+    responses:
+        200:
+            description: "Successfully logged in"
+        401:
+            description: "Failed to login"
+
+    """
+    user = User.query.filter_by(username=request.data['username']).first()
+
+    if user and user.password_is_valid(request.data['password']):
+        auth_token = user.encode_auth_token(user.id)
+        if auth_token:
+            response = {
+                'message': 'Succesfully logged in',
+                'status': 'Success',
+                'auth_token': auth_token.decode()
+            }
+            return make_response(jsonify(response)), 200
+    response = {
+        'message': 'Failed to login, unknown username or password',
+        'status': 'Failed'
+    }
+    return make_response(jsonify(response)), 401
+
+@auth_blueprint.route('/reset-password', methods=['POST'])
+def reset_password():
+    """Reset User password
+    ---
+    tags:
+     - "auth"
+    parameters:
+        - in: "header"
+          name: "Authorization"
+          description: "Token of logged in user"
+          required: true
+        - in: "body"
+          name: "body"
+          description: "Username, old password and new password"
+          required: true
+          schema:
+            type: "object"
+            required:
+            - "username"
+            - "old_password"
+            - "new_password"
+            properties:
+                username:
+                    type: "string"
+                old_password:
+                    type: "string"
+                new_password:
+                    type: "string"
+    responses:
+        200:
+            description: "Successfully changed password"
+        400:
+            description: "Failed to reset password, bad username or password"
+    """
+    user = User.query.filter_by(username=request.data['username']).first()
+
+    if user and user.password_is_valid(request.data['old_password']):
+        user.password = Bcrypt().generate_password_hash(password=request.data['new_password'])
+        user.save()
+        response = {
+            'message': 'Successfully changed password',
+            'status': 'Success'
+        }
+        return make_response(jsonify(response)), 200
+    response = {
+        'message': 'Failed to reset password, bad username or password',
+        'status': 'Failed'
+    }
+    return make_response(jsonify(response)), 400
```

Comment:
should be 401. 400 is for bad request
https://en.wikipedia.org/wiki/List_of_HTTP_status_codes